### DISCUSSION TITLE
Need help，how to trigger 'Data Provider' update after Table Row edit

### DISCUSSION THREAD

**Discussion by yoyogis** [Budibase/budibase#14584] [Help] [answered]
I created a screen with a table and a pie chart. Their data points to the same DB table. After I change the value in a table cell, how can I trigger the "Data Provider" to update and refresh the pie chart?
<img width="1152" alt="image" src="https://github.com/user-attachments/assets/2985a564-a883-412b-85f4-28c235aa0e79">

**Comment by mikesealey** [comment, marked as answer]
Hey @yoyogis 

In the Table we can do inline editing, but when the row is saved we don't have the option to run any actions. For this scenario you might be better editing a row in a modal or a side panel. 

Add a button per row in your table
![image](https://github.com/user-attachments/assets/2a6ae822-7e42-4429-b3ca-dce146a00b54)

With two actions set - update state, and open modal
![image](https://github.com/user-attachments/assets/a91a8f68-e4d2-4376-8102-5fa7bcf5a962)

You'll need to add a modal component, and set the action to open it.

Then inside your modal, add a form block set to update, and give it the state value where you're storing your row id.
![image](https://github.com/user-attachments/assets/d1b9dd15-2a4f-48db-8d9f-ddd7ed4c431e)

In the Form Block's save button action, make sure to add a "refresh data provider"
![image](https://github.com/user-attachments/assets/d291c545-f64c-4255-a2ec-a587f625f5d3)

Here's an app-export where you can take a closer look at my work.
[14584-export-1727106932183.tar.gz](https://github.com/user-attachments/files/17100471/14584-export-1727106932183.tar.gz)

**Comment by phamalda** [comment]
@yoyogis, this is a really cool pie chart. I have been working on trying to successfully create one for a while. Would you mind sharing with me how you built that chart? I would really appreciate it.
